ql/src/java/org/apache/hadoop/hive/ql/ddl/table/constraint/ConstraintsUtils.java

```diff
@@ -309,8 +312,8 @@ private static List<ConstraintInfo> generateConstraintInfos(ASTNode child, List<
    * @param node AST node corresponding to default value
    * @return retrieve the default value and return it as string
    */
-  private static Pair<TypeInfo, String> getDefaultValueAndType(ASTNode node, TokenRewriteStream tokenStream)
-      throws SemanticException{
+  private static Pair<TypeInfo, String> getDefaultValueAndType(ASTNode node, TokenRewriteStream tokenStream,
+      boolean isNegativeConstant) throws SemanticException {
     // first create expression from defaultValueAST
     TypeCheckCtx typeCheckCtx = new TypeCheckCtx(null);
     ExprNodeDesc defaultValExpr = ExprNodeTypeCheck.genExprNode(node, typeCheckCtx).get(node);
```

Comment:
The function `ExprNodeTypeCheck.genExprNode` is able to parse AST trees like
```
-
    15
```
So the ASTNode represents the `-` sign is the parent of the ASTNode represents the number literal `15`.
The parameter `isNegativeConstant` is unnecessary.